src/main.rs

```diff
@@ -1,159 +1,226 @@
-extern crate reqwest;
 extern crate dotenv;
 extern crate mongodb;
+extern crate reqwest;
+#[macro_use]
+extern crate prettytable;
+
 pub mod protocol;
-use protocol::*;
+
 use dotenv::dotenv;
-use std::{env, io};
+use protocol::*;
 use reqwest::StatusCode;
+use std::{env, io};
 //use chrono::{DateTime, FixedOffset};
-use mongodb::{Bson, bson, doc, Client, ThreadedClient};
 use mongodb::db::ThreadedDatabase;
-#[macro_use] extern crate prettytable;
+use mongodb::{bson, doc, Bson, ThreadedClient};
 use prettytable::Table;
 
-fn main() -> Result<(), Box<std::error::Error>> {
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
```

Comment:
This is a nice trick for cutting down repetitive types. Since all of your `Result`s in this file have the same Err type, you can shorten it by pre-filling the error type. Now in the rest of the file, you only have one type argument for your results, which is the "successful" type.

E.g., with `fn main() -> Result<()>`, the full return type is still `Result<(), Box<dyn std::error:Error>>`, but you only have to type `Result<()>` :+1:

Reply:
:O Yo that's insane